pkg/log/zap/zap.go

```diff
@@ -132,6 +134,8 @@ type Options struct {
 	// Encoder configures how Zap will encode the output.  Defaults to
 	// console when Development is true and JSON otherwise
 	Encoder zapcore.Encoder
+	// TimeEncoder configures the zap time format. Defaults to epoch.
+	TimeEncoder zapcore.TimeEncoder
```

Comment:
@bharathi-tenneti How would I configure this option with `zap.New()` without using any command line flags?

We have functional options for all the other fields that can be configured without flags:
```Go
log := NewRaw(zap.UseDevMode(true), zap.WriteTo(destWriter), zap.RawZapOpts(zapOpts...))
```

However if we just add the functional option helper for TimeEncoder to set the option as is:
```Go
func TimeEncoder(timeEncoder zapcore.TimeEncoder) func(o *Options) {
	return func(o *Options) {
		o.TimeEncoder = timeEncoder
	}
}
```
It could be a little cumbersome to set with `zap.New()`:
```Go
var te zapcore.TimeEncoder
if err := te.UnmarshalText([]byte("millis")); err != nil {
	return fmt.Errorf("not able to unmarshal time encoder value \"%s\"", val)
}

log := zap.New(zap.UseDevMode(true), zap.TimeEncoder(te))

```

So in the interest of usability I'm wondering if we can't change this field to be `TimeEncoder string`, so that the option is easier to set:
```Go
func TimeEncoder(timeEncoder string) func(o *Options) {
	return func(o *Options) {
		o.TimeEncoder = timeEncoder
	}
}

log := zap.New(zap.UseDevMode(true), zap.TimeEncoder("millis"))
```
And `addDefaults()` is the place where the TimeEncoder is actually initialized based on the option string:
```Go
func (o *Options) addDefaults() {
...
	if o.Development {
		if o.Encoder == nil {
			if o.TimeEncoder != "" {
				ecfs = append(ecfs, withTimeEncoding(o.TimeEncoder))
			}
			o.Encoder = newConsoleEncoder(ecfs)
		}
...
}

func withTimeEncoding(timeEncoderFormat string) encoderConfigFunc {
	return func(ec *zapcore.EncoderConfig) {
		var te zapcore.TimeEncoder
		if err := te.UnmarshalText([]byte(timeEncoderFormat)); err != nil {
			// As of zap v1.9.1 TimeEncoder's `UnmarshalText()` does not return an error and defaults to epoch
			// but if it ever does, default to epoch.
			te.UnmarshalText([]byte("epoch"))
		}
		ec.EncodeTime = te
	}
}
```

`BindFlags()` would need to be updated slightly but it should still be doable I think.

Reply:
@hasbro17 @bharathi-tenneti I think `zapcore.Encoder` is probably okay.

It isn't necessary to use `UnmarshalText`. The zapcore package contains a bunch of named and exported TimeEncoder implementations (e.g. [`zapcore.EpochMillisTimeEncoder`](https://godoc.org/go.uber.org/zap/zapcore#EpochMillisTimeEncoder))

So if I'm understanding correctly, a user could do this:
```go
te := zapcore.EpochMillisTimeEncoder
log := zap.New(zap.UseDevMode(true), zap.TimeEncoder(te))
```